### DISCUSSION TITLE
How to check format of a response parameter

### DISCUSSION THREAD

**Discussion by AnjaliVerma1203** [thunderclient/thunder-client-support#1455] [Q&A] [answered]
Hello,

We are exploring Thunder client for our testing needs and would like to test 2 cases:
1. check if update_date parameter in below response is in format YYYY-MM-DD 
2. check if score parameter display non decimal numbers between 0-5
Below is the Json structure():
![Untitled](https://github.com/rangav/thunder-client-support/assets/156674794/fad93ea6-c12c-4d33-a78c-60f7cf93a00f)
Please guide very new to scripting or if this can be done in Tests tab.

**Comment by rangav** [comment]
Hi @AnjaliVerma1203 

you can use scripting in Tests tab and write test cases

you need to write your own javascript logic 

see example below
https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#assertions-using-scripting

**Comment by rangav** [comment]
Are you using free version or paid version of the extension?

**Reply by AnjaliVerma1203** [reply]
Currently using free version to check if it fulfil our testing needs.

Wrote below logic to test score parameter but its showing error "tc.response.json is not a function", pls suggest I have little to no experience with javascript.

![image](https://github.com/rangav/thunder-client-support/assets/156674794/b0bf4efa-077f-4e73-b4da-24278eb182bc)

**Reply by AnjaliVerma1203** [reply]
Updated javascript to this and now its giving error: "undefined" is not valid JSON.

content-type is application/json. This test is passed

![Untitled](https://github.com/rangav/thunder-client-support/assets/156674794/8057383c-82b3-48c2-984e-343baf8fc16c)

**Reply by rangav** [reply, marked as answer]
Use this correct statement `const response = tc.response.json`

**Reply by AnjaliVerma1203** [reply]
already tried its giving error "tc.response.json is not a function"

**Reply by rangav** [reply]
No i think you used tc.response.json() with braces which is wrong

**Reply by AnjaliVerma1203** [reply]
Thanks it resolved the script error. But TC is failed 

![image](https://github.com/rangav/thunder-client-support/assets/156674794/18801f15-b15a-49f4-aa7d-a97d028be2cc)

**Reply by rangav** [reply]
The json query you wrote is wrong I think, use stackoverflow or ask help from your colleagues

**Reply by AnjaliVerma1203** [reply]
Sure, thanks for your help
